main.go

```diff
@@ -190,6 +203,163 @@ func main() {
+		resp, err := http.Get("http://localhost:8081/readyz")
+		if err != nil {
+			setupLog.Error(err, "unable to do ready check")
+			os.Exit(1)
+		}
+
+		if resp.StatusCode == http.StatusOK {
+			setupLog.Info("controller is ready")
+			resp.Body.Close()
+
+			break
+		}
+
+		setupLog.Info("controller is not ready, retrying...")
+		resp.Body.Close()
+	}
+
+	ctx := context.TODO()
+
+	lease := &coordv1.Lease{}
+	leaseNamespacedName := types.NamespacedName{
+		Name:      leaderElectionID,
+		Namespace: os.Getenv("POD_NAMESPACE"),
+	}
+
+	if err := client.Get(ctx, leaseNamespacedName, lease); err != nil {
+		setupLog.Error(err, "unable to get lease object")
+		os.Exit(1)
+	}
+
+	// Migration steps should be done by one pod only.
+	if !strings.HasPrefix(*lease.Spec.HolderIdentity, os.Getenv("POD_NAME")) {
+		setupLog.Info("HolderIdentity not matching", "podname",
+			os.Getenv("POD_NAME"), "holderIdentity", *lease.Spec.HolderIdentity)
+		return
+	}
+
+	setupLog.Info("Formatting Initialised Volumes name")
+
+	if err := addNewlyFormattedInitialisedVolumeNames(ctx, client, setupLog); err != nil {
+		setupLog.Error(err, "problem patching Initialised volumes")
+		os.Exit(1)
+	}
+}
+
+func addNewlyFormattedInitialisedVolumeNames(ctx context.Context, client crClient.Client, setupLog logr.Logger) error {
+	aeroClusterList := &asdbv1.AerospikeClusterList{}
+	if err := client.List(ctx, aeroClusterList); err != nil {
+		if errors.IsNotFound(err) {
+			// Request objects not found.
+			return nil
+		}
+		// Error reading the object.
+		return err
+	}
+
+	for idx := range aeroClusterList.Items {
+		aeroCluster := aeroClusterList.Items[idx]
+
+		podList, err := getClusterPodList(ctx, client, &aeroCluster)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Request objects not found.
+				return nil
+			}
+			// Error reading the object.
+			return err
+		}
+
+		var patches []jsonpatch.PatchOperation
+
+		for podIdx := range podList.Items {
+			pod := podList.Items[podIdx]
+			initializedVolumes := aeroCluster.Status.Pods[pod.Name].InitializedVolumes
+			initializedVolumesMap := make(map[string]string)
+			oldFormatInitVolNames := make([]string, 0, len(initializedVolumes))
+
+			for idx := range initializedVolumes {
+				initVolInfo := strings.Split(initializedVolumes[idx], "@")
+				if len(initVolInfo) < 2 {
+					oldFormatInitVolNames = append(oldFormatInitVolNames, initializedVolumes[idx])
+				} else {
+					initializedVolumesMap[initVolInfo[0]] = initVolInfo[1]
+				}
+			}
+
+			for idx := range oldFormatInitVolNames {
+				if _, ok := initializedVolumesMap[oldFormatInitVolNames[idx]]; !ok {
+					pvcUID, pvcErr := getPVCUid(context.TODO(), client, &pod, initializedVolumes[idx])
+					if pvcErr != nil {
+						return pvcErr
+					}
+
+					// Appending volume name as <vol_name>@<pvcUID> in initializedVolumes list
+					initializedVolumes = append(initializedVolumes, fmt.Sprintf("%s@%s", initializedVolumes[idx], pvcUID))
+				}
+			}
```

Comment:
These 2 can be combined in 1 for loop

Reply:
Combining these loops is not possible, but I changed the initializedVolumesMap to sets of string.